PIL/WebPImagePlugin.py

```diff
@@ -30,17 +84,109 @@ class WebPImageFile(ImageFile.ImageFile):
     format_description = "WebP image"
 
     def _open(self):
-        data, width, height, self.mode, icc_profile, exif = \
-            _webp.WebPDecode(self.fp.read())
 
-        if icc_profile:
-            self.info["icc_profile"] = icc_profile
-        if exif:
-            self.info["exif"] = exif
+        header = self.fp.read(12)
+        if 12 != len(header):
+            raise SyntaxError("not a WebP file")
+
+        if b'RIFF' != header[0:4] or b'WEBP' != header[8:12]:
+            raise SyntaxError("not a WebP file")
+
+        mode = None
+        size = None
+        lossy = True
+
+        first_chunk = True
+
+        while True:
+
+            chunk_header = self.fp.read(8)
+            if len(chunk_header) < 8:
+                # EOF.
+                break
+
+            chunk_fourcc = chunk_header[0:4]
+            chunk_size = i32le(chunk_header[4:8])
+
+            if first_chunk:
+                if chunk_fourcc not in _VP8_MODES_BY_IDENTIFIER:
+                    raise SyntaxError("not a WebP file")
+
+            if b'VP8X' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8x(chunk_size, self.fp)
+
+            elif b'VP8L' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8l(chunk_size, self.fp)
+                lossy = False
+
+            elif b'VP8 ' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8(chunk_size, self.fp)
+                lossy = True
```

Comment:
May be it should be None by default, because there is a check for None in line 148 below.